cards for ttZ plain NLO

Cvico opened the pull request:
Hi,

These are the cards for plain ttZ NLO (dileptonic final states). Details:

1. Removed FxFx stuff
2. Set mmaxgamma to 4... start to be consistent now with this cut for future samples?

Best,
Carlos

sihyunjeon commented on Cards/MadGraph5_aMCatNLO/TT/TTLL_MLL-10to50_amcatnlo-pythia8/TTLL_MLL-10to50_amcatnlo-pythia8.json:
```diff
@@ -0,0 +1,23 @@
+        "set run_card mll_sf 10.0"
+    ],
+    "model_params_vars": {},
+    "fragment": ["Generator/ExternalLHEProducer.dat", "PartonShower/amcatnloFXFX-pythia8.dat"],
```
This should be using plain amcatnlo-pythia8.dat

Cvico replied:
Oops I missed this one, updated!

sihyunjeon commented on Cards/MadGraph5_aMCatNLO/TT/TTLL_MLL-10to50_amcatnlo-pythia8/TTLL_MLL-10to50_amcatnlo-pythia8.json:
```diff
@@ -0,0 +1,23 @@
+    "fragment_user": [],
+    "fragment_vars": {
+        "processParameters": [
+            "'TimeShower:nPartonsInBorn = 2'",
```
Ah I totally forgot about this.
@agrohsje confirm please : for plain amcatnlo samples that are not using FXFX, we should specify this right?

agrohsje replied:
Yes. It should be specified. I checked and the logic only works when there is a npNLO key word which we don't have in plan MC@NLO.

sihyunjeon commented:
One question
is `/ h` necessary?

Cvico commented:
> 

I think we discussed this at some point. ` / h` is hand-by-hand removing the ttH->tau tau interference with ttZ->tau tau so I think we ended up keeping it

sihyunjeon commented:
ah i think we did uhmmm the trickiest thing ok.
Now thinking a bit more we could've actually turn off the yukawa coupling of taus and make it give 0 ME amplitude in customize card.
Let's keep this in mind (perhaps for next round).

agrohsje commented:
About cuts.f. We should handle this as a patch to the version-dependent cuts.f. @sihyunjeon should we enable this logic now?

sihyunjeon commented:
> About cuts.f. We should handle this as a patch to the version-dependent cuts.f. @sihyunjeon should we enable this logic now?

But how should we make this work?
cuts.f runs over multiple loops here and there so I am not 100% sure how we can create the template and make a replacable

Cvico commented:
> > About cuts.f. We should handle this as a patch to the version-dependent cuts.f. @sihyunjeon should we enable this logic now?
> 
> But how should we make this work? cuts.f runs over multiple loops here and there so I am not 100% sure how we can create the template and make a replacable

I think this depends on how much the loop previous to the one with the invariant mass cut changes... For example in 2.6.5 the passcuts_user was named passcuts_lepton iirc. So these kinds of minimal changes I think could end up in making very troublesame to write a template for all the future cases. Will try to think of something anyway

agrohsje commented:
Not sure I understand. What is the problem with a patch. I mean 
diff -ua cuts.f cuts_modified.f and then apply this ?

Cvico commented:
> Not sure I understand. What is the problem with a patch. I mean diff -ua cuts.f cuts_modified.f and then apply this ?

Oh I thought the plan was to have replaceables in the cuts.f. What difference does it make to have patches like that over replacing the cuts.f file after untaring the MG release? Just to understand fully what's the plan

agrohsje commented:
My point is that we know cuts.f can change with the version. However, usually, the blog of cuts is the same, as variable names are the same etc. So in this case the patch just contains the snippet with our 5-6 lines of code. So no matter what happens with the rest of cuts.f, the snippet can still be inserted.

Cvico commented:
> My point is that we know cuts.f can change with the version. However, usually, the blog of cuts is the same, as variable names are the same etc. So in this case the patch just contains the snippet with our 5-6 lines of code. So no matter what happens with the rest of cuts.f, the snippet can still be inserted.

I see your point now, I was not understanding. I've updated the PR with the patches for each mass bin just to bring up discussion, do you think this should be done with a replacer? something along the lines:

```
@@ -138,6 +138,13 @@
                               passcuts_user=.false.
                               return
                            endif
+                        endif
+                        if (mll_sf.gt.0d0) then
+                           if (invm2_04(p(0,i),p(0,j),1d0).gt.$mass**2)
+     $                          then
+                              passcuts_user=.false.
+                              return
+                           endif
                         endif
                         if (mll_sf.gt.0d0) then
                            if (invm2_04(p(0,i),p(0,j),1d0).lt.mll_sf**2)
```

Note I've added "$mass" in the middle (when the cut is actually applied). 
Do you think this is an overkill or something worth doing?

Cvico commented:
Last commit adds the `a/` and `b/` to the headers so the file gets properly patched
